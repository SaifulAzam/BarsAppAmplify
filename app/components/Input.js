import PropTypes from 'prop-types';
import React from 'react';
import {
  View, TextInput, StyleSheet, Dimensions,
} from 'react-native';

const { width } = Dimensions.get('window');

const Input = ({
  onChangeText, value, placeholder, secureTextEntry, style,
}) => (
  <View>
    <TextInput
      placeholder={placeholder}
      returnKeyType="done"
      underlineColorAndroid="transparent"
      style={[styles.inputStyle, style]}
      onChangeText={onChangeText}
      value={value}
      secureTextEntry={secureTextEntry}
      autoCorrect={false}
      autoCapitalize="none"
    />
  </View>
);

const styles = StyleSheet.create({
  inputStyle: {
    backgroundColor: 'white',
    marginVertical: 10,
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 10,
    paddingRight: 10,
    width: width * 0.8,
  },
});

export default Input;

Input.propTypes = {
  onChangeText: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  placeholder: PropTypes.string.isRequired,
  secureTextEntry: PropTypes.bool,
  style: PropTypes.shape(),
};

Input.defaultProps = {
  value: null,
  secureTextEntry: false,
  style: {},
};
