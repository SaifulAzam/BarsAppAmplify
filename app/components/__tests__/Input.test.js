import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

test('renders correctly', () => {
  const tree = renderer
    .create(<Input placeholder="test@test.com" onChangeText={() => console.log('Text changed.')} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
