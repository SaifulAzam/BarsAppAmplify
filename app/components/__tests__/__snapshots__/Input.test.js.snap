// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="test@test.com"
    returnKeyType="done"
    secureTextEntry={false}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "marginVertical": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": 600,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    value={null}
  />
</View>
`;
